sql/parser/parse_test.go

```diff
@@ -275,9 +275,14 @@ func TestParse2(t *testing.T) {
 			`SELECT CAST('1' AS INT)`},
 		// Double negation. See #1800.
 		{`SELECT *,-/* comment */-5`,
-			`SELECT *, - -5`},
+			`SELECT *, - - 5`},
 		{"SELECT -\n-5",
-			`SELECT - -5`},
+			`SELECT - - 5`},
+		{"SELECT 1e-\n-1",
```

Comment:
`1e-` doesn't appear to be commonly accepted; both postgres and python require an exponent value (`1e-0`).